packages/reactivity/src/reactive.ts

```diff
@@ -141,7 +141,7 @@ export function shallowReactive<T extends object>(
 }
 
 type Primitive = string | number | boolean | bigint | symbol | undefined | null
-type Builtin = Primitive | Function | Date | Error | RegExp
+export type Builtin = Primitive | Function | Date | Error | RegExp
```

Comment:
This isn't correct because the readonly vs. ref unwrap logic are not entirely the same. `Date | Error | RegExp` are valid ref values and should be unwrapped.

Instead, we can export the `Primitive` type and use it instead of the old `BaseType`.